app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/SearchPopover.vue

```diff
@@ -0,0 +1,165 @@
+    >
+      <search-header
+        :title="$t('HELP_CENTER.ARTICLE_SEARCH.TITLE')"
+        class="w-full sticky top-0 bg-[inherit]"
+        @close="onClose"
+        @search="onSearch"
+      />
+
+      <article-view
+        v-if="activeId"
+        :url="articleViewerUrl"
+        @back="onBack"
+        @insert="onInsert"
+      />
+      <search-results
+        v-else
+        :search-query="searchQuery"
+        :is-loading="isLoading"
+        :portal-slug="selectedPortalSlug"
+        :articles="searchResultsWithUrl"
+        @preview="handlePreview"
+        @insert="onInsert"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import { debounce } from '@chatwoot/utils';
+import { mixin as clickaway } from 'vue-clickaway';
+import {
+  isEscape,
+  isActiveElementTypeable,
+} from 'shared/helpers/KeyboardHelpers';
+
+import SearchHeader from './Header.vue';
+import SearchResults from './SearchResults.vue';
+import ArticleView from './ArticleView.vue';
+import ArticlesAPI from 'dashboard/api/helpCenter/articles';
+import portalMixin from '../../mixins/portalMixin';
+
+export default {
+  name: 'ArticleSearchPopover',
+  components: {
+    SearchHeader,
+    SearchResults,
+    ArticleView,
+  },
+  mixins: [clickaway, portalMixin],
+  props: {
+    selectedPortalSlug: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      searchQuery: '',
+      isLoading: false,
+      searchResults: [],
+      activeId: '',
+      debounceSearch: () => {},
+    };
+  },
+  computed: {
+    articleViewerUrl() {
+      const article = this.activeArticle(this.activeId);
+      if (!article) return undefined;
+      const isDark = document.body.classList.contains('dark');
+
+      const url = new URL(article.url);
+      url.searchParams.set('show_plain_layout', 'true');
+
+      if (isDark) {
+        url.searchParams.set('theme', 'dark');
+      }
+
+      return `${url}`;
+    },
+    searchResultsWithUrl() {
+      return this.searchResults.map(article => ({
+        ...article,
+        localeName: this.localeName(article.category.locale || 'en'),
+        url: this.generateArticleUrl(article),
+      }));
+    },
+  },
+  mounted() {
+    // Load popular articles on mount
+    this.fetchArticlesByQuery(this.searchQuery);
+    this.debounceSearch = debounce(this.fetchArticlesByQuery, 500, true);
+    document.body.addEventListener('keydown', this.closeOnEsc);
+  },
+  beforeDestroy() {
+    document.body.removeEventListener('keydown', this.closeOnEsc);
+  },
+  methods: {
+    generateArticleUrl(article) {
+      const host =
+        window.chatwootConfig.helpCenterURL || window.chatwootConfig.hostURL;
+      return `${host}/hc/${this.selectedPortalSlug}/articles/${article.slug}`;
```

Comment:
This is used somewhere else in the codebase, please create a helper.

Reply:
Done